### PR TITLE
Changes for #1567

### PR DESCRIPTION
**Pull request by jgustie**
This includes changes I mentioned in the ticket along with a couple of minor fixes (commits are isolated).

### CODE REVIEW

**Comment by jgustie** [comment]
CLA comment bump.

File: pkg/transformers/config/transformerconfig.go [hunk 1]
```diff
@@ -56,6 +56,7 @@ func MakeDefaultConfig() *TransformerConfig {
 // sortFields provides determinism in logging, tests, etc.
 func (t *TransformerConfig) sortFields() {
 	sort.Sort(t.NamePrefix)
+	sort.Sort(t.NameSuffix)
```
Comment (jbrette) [inline pkg/transformers/config/transformerconfig.go]: This is irrelevant. FieldSpecs related to suffix (t.NameSuffix) should have been deprecated long time ago. The transformer only uses [one set](https://github.com/kubernetes-sigs/kustomize/blob/master/plugin/builtin/prefixsuffixtransformer/PrefixSuffixTransformer.go#L21).

So when the transformer is initialized, it only passes one t.NamePrefix: [here](https://github.com/kubernetes-sigs/kustomize/blob/master/pkg/target/kusttarget_configplugin.go#L251)

File: pkg/gvk/gvk.go [hunk 2]
```diff
@@ -117,12 +117,21 @@ var typeOrders = func() map[string]int {
...
+		return x.Kind < o.Kind
+	}
+	if x.Group != o.Group {
+		return x.Group != "" && (o.Group == "" || x.Group < o.Group)
+	}
+	if x.Version != o.Version {
+		return x.Version != "" && (o.Version == "" || x.Version < o.Version)
 	}
-	return x.String() < o.String()
+	return false
```
Comment (jgustie) [inline pkg/gvk/gvk.go]: It just occurred to me that a less intrusive approach would have been to just swap the usage of `i` and `j` in `fsSlice.Less`). I also didn't add a sort on `fsSlice.mergeOne` after the `append` so individual configs could still be out of order.

File: pkg/transformers/config/transformerconfig.go (hunk 1 above)
Comment (jgustie) [reply pkg/transformers/config/transformerconfig.go]: Sorry, I wasn't aware of the semantics, it was mostly just convenient to use from the existing test and I initially got unexpected results without it.

**Comment by jgustie** [comment]
I'm closing this, it does not address the issue I am seeing.

**Pull request closed without merging**